src/Data/Foreign.purs

```diff
@@ -68,20 +69,18 @@ renderForeignError (ForeignError msg) = msg
 renderForeignError (ErrorAtIndex i e) = "Error at array index " <> show i <> ": " <> show e
 renderForeignError (ErrorAtProperty prop e) = "Error at property " <> show prop <> ": " <> show e
 renderForeignError (JSONError s) = "JSON error: " <> s
-renderForeignError (TypeMismatch exps act) = "Type mismatch: expected " <> listTypes exps <> ", found " <> act
-  where
-  listTypes (NE.NonEmpty typ []) = typ
-  listTypes typs = "one of " <> joinWith ", " (NE.oneOf typs)
+renderForeignError (TypeMismatch exp act) = "Type mismatch: expected " <> exp <> ", found " <> act
 
--- | An error monad, used in this library to encode possible failure when
+-- | An error monad, used in this library to encode possible failures when
 -- | dealing with foreign data.
-type F = Either ForeignError
+type F = Either (NE.NonEmpty List ForeignError)
```

Comment:
Ah right, I was trying to avoid adding the `-transformers` dependency, but yep, good point.